tests/functional/test_import.py

```diff
@@ -47,14 +47,105 @@ def test_ctypes_CDLL_c(pyi_builder):
+                 (defaults to 'lib%s.so' % libname)
+
+    :return: pytest decorator with a reason.
+    """
+    soname = ctypes.util.find_library(libname)
+    if not text:
+        text = "lib%s.so" % libname
+    # Return pytest decorator.
+    return skipif(not (soname and ctypes.CDLL(soname)),
+                  reason="required %s missing" % text)
+
+
+_template_ctypes_CDLL_find_library = """
+    import ctypes, ctypes.util, sys, os
+    lib = ctypes.CDLL(ctypes.util.find_library(%(libname)r))
+    print(lib)
+    assert lib is not None and lib._name is not None
+    if getattr(sys, 'frozen', False):
+        soname = ctypes.util.find_library(%(libname)r)
+        print(soname)
+        libfile = os.path.join(sys._MEIPASS, soname)
+        print(libfile)
+        assert os.path.isfile(libfile), '%%s is missing' %% soname
+        print('>>> file found')
+    """
+
+_template_ctypes_test = """
+        print(lib)
+        assert lib is not None and lib._name is not None
+        import sys, os
+        if getattr(sys, 'frozen', False):
+            libfile = os.path.join(sys._MEIPASS, %(soname)r)
+            print(libfile)
+            assert os.path.isfile(libfile), '%(soname)s is missing'
+            print('>>> file found')
+    """
+
+# Ghostscript's libgs.so should be available in may Unix/Linux systems
+@skip_if_lib_missing('gs', 'libgs.so (Ghostscript)')
+def test_ctypes_CDLL_find_library__gs(pyi_builder):
+    libname = 'gs'
+    pyi_builder.test_source(_template_ctypes_CDLL_find_library % locals())
+
+
+#-- Generate test-cases for the different types of ctypes objects.
+
+libname = 'gs'
+reason = 'libgs.so (Ghostscript)'
+parameters = []
+ids = []
+for prefix in ('', 'ctypes.'):
+    for funcname in  ('CDLL', 'PyDLL', 'WinDLL', 'OleDLL', 'cdll.LoadLibrary'):
+        ids.append('%s_%s' % (prefix+funcname, libname))
+        params = (prefix+funcname, libname, reason, ids[-1])
+        if funcname in ("WinDLL", "OleDLL"):
+            # WinDLL, OleDLL only work on windows.
+            params = skipif_notwin(params)
+        parameters.append(params)
+
+@pytest.mark.parametrize("funcname,libname,reason,test_id", parameters, ids=ids)
+@skip_if_lib_missing(libname, reason)
+def test_ctypes_gen(pyi_builder, funcname, libname, reason, test_id):
+    # evaluate the soname here, so the test-code contains a constant
+    soname = ctypes.util.find_library(libname)
+    source = """
+        import ctypes ; from ctypes import *
+        lib = %s(%%(soname)r)
+    """ % funcname + _template_ctypes_test
+    source = source +_template_ctypes_test
+    pyi_builder.test_source(source % locals(), test_id=test_id)
+
+
+# TODO: Add test-cases forthe prefabricated library loaders supporting
+# attribute accesses on windows. Example::
+#
+#   cdll.kernel32.GetModuleHandleA(None)
+#
+# Of course we need to use dlls which is not are commony available on
+# windows but mot excluded in PyInstaller.depend.dylib
```

Comment:
Roger that. I'll try to find a DLL we can use on AppVeyor. The excludes list should be expanded to have all system .DLLs, too.